src/audio/codec_adapter/codec/dummy.c

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright 2020 NXP
+//
+// Author: Daniel Baluta <daniel.baluta@nxp.com>
+//
+// Dummy codec implementation to demonstrate Codec Adapter API
+
+#include <sof/audio/codec_adapter/codec/generic.h>
+#include <sof/audio/codec_adapter/codec/dummy.h>
+
+int dummy_codec_init(struct comp_dev *dev)
+{
+	comp_info(dev, "dummy_codec_init() start");
+	return 0;
+}
+
+int dummy_codec_prepare(struct comp_dev *dev)
+{
+	struct codec_data *codec = comp_get_codec(dev);
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	comp_info(dev, "dummy_codec_process()");
+
+	codec->cpd.in_buff = rballoc(0, SOF_MEM_CAPS_RAM, cd->period_bytes);
+	if (!codec->cpd.in_buff) {
+		comp_err(dev, "dummy_codec_prepare(): Failed to alloc in_buff");
+		return -ENOMEM;
+	}
+	codec->cpd.in_buff_size = cd->period_bytes;
+
+	codec->cpd.out_buff = rballoc(0, SOF_MEM_CAPS_RAM, cd->period_bytes);
+	if (!codec->cpd.out_buff) {
+		comp_err(dev, "dummy_codec_prepare(): Failed to alloc out_buff");
+		return -ENOMEM;
```

Comment:
In case out_buff cannot be allocated, in_buff shouldn't be freed? Or the dummy_codec_free() is always called?

Reply:
@iuliana-prodan you are right. I have updated the code.